sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -0,0 +1,388 @@
+     *
+     * @return A new AMQP message for this {@code object}.
+     *
+     * @throws IllegalArgumentException if {@code object} is not an instance of {@link Message}.
+     */
+    @Override
+    public <T> org.apache.qpid.proton.message.Message serialize(T object) {
+        Objects.requireNonNull(object, "'object' to serialize cannot be null.");
+
+        if (!(object instanceof Message)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "Cannot serialize object that is not EventData. Clazz: " + object.getClass()));
+        }
+
+        final Message eventData = (Message) object;
+        final org.apache.qpid.proton.message.Message message = Proton.message();
+
+        if (eventData.getProperties() != null && !eventData.getProperties().isEmpty()) {
+            message.setApplicationProperties(new ApplicationProperties(eventData.getProperties()));
+        }
+
+        setSystemProperties(eventData, message);
+
+        message.setBody(new Data(new Binary(eventData.getBody())));
+
+        return message;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> T deserialize(org.apache.qpid.proton.message.Message message, Class<T> clazz) {
+        Objects.requireNonNull(message, "'message' cannot be null.");
+        Objects.requireNonNull(clazz, "'clazz' cannot be null.");
+
+        if (clazz == ServiceBusProperties.class) {
+            return deserializeManagementResponse(message, clazz);
+        } else if (clazz == Message.class) {
+            return (T) deserializeEventData(message);
+        } else {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "Deserialization only supports ServiceBusProperties."));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T deserializeManagementResponse(org.apache.qpid.proton.message.Message message, Class<T> deserializedType) {
+        if (!(message.getBody() instanceof AmqpValue)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "Expected message.getBody() to be AmqpValue, but is: " + message.getBody()));
+        }
+
+        final AmqpValue body = (AmqpValue) message.getBody();
+        if (!(body.getValue() instanceof Map)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "Expected message.getBody().getValue() to be of type Map"));
+        }
+
+        final Map<?, ?> amqpBody = (Map<?, ?>) body.getValue();
+
+        if (deserializedType == ServiceBusProperties.class) {
+            return (T) toServiceBusProperties(amqpBody);
+        } else {
+            throw logger.logExceptionAsError(new IllegalArgumentException(String.format(
+                Messages.CLASS_NOT_A_SUPPORTED_TYPE, deserializedType)));
+        }
+    }
+
+
+    private Message deserializeEventData(org.apache.qpid.proton.message.Message message) {
```

Comment:
deserializeEventData -> deserializeMessage. Does this match the track 1 deserialization logic?

Reply:
I will check for track 1 deserialization .